packages/beacon-node/src/chain/blocks/importBlock.ts

```diff
@@ -414,6 +386,14 @@ export async function importBlock(
 
   if (this.clock.currentSlot - blockSlot < EVENTSTREAM_EMIT_RECENT_BLOCK_SLOTS) {
     // NOTE: Skip looping if there are no listeners from the API
+
+    if (this.emitter.listenerCount(routes.events.EventType.block)) {
```

Comment:
when node has connected validators we need to import block asap
need to follow the old implementation to emit all events in the next event loop
```typescript
 // We want to import block asap so call all event handler in the next event loop
  callInNextEventLoop(async () => {
```

Reply:
at least NetworkProcessor listens for block event to process attestations here https://github.com/ChainSafe/lodestar/blob/42af692453b575621d3ba0ce928c4bea0c6bfbfd/packages/beacon-node/src/network/processor/index.ts#L184

guard the block with `callInNextEventLoop` and we don't have to worry about any new event handlers in the future which could delay the block import